src/com/sap/piper/GitUtils.groovy

```diff
@@ -43,21 +43,18 @@ String getGitMergeCommitId(String gitChangeId){
 }
 
 boolean compareParentsOfMergeAndHead(String mergeCommitId){
-    try {
-        String mergeCommitParents = sh(returnStdout: true, script: "git rev-parse ${mergeCommitId}^@ | tac").trim()
-        String headCommitParents = sh(returnStdout: true, script: "git rev-parse HEAD^@").trim()
-        if(mergeCommitParents.equals(headCommitParents)){
-            return true
-        }
-    } catch (Exception e) {
-        echo 'Error comparing merge commit parents and local merge commit parents'
-        throw e
+    if(!pullRequest){
+        echo 'pullRequest context not found'
+        throw new Exception('pullRequest context not found')
     }
 
+    if( pullRequest.mergeCommitSha.equals(mergeCommitId) ){
+        return true
+    }
 
-    echo "GH merge parents: ${mergeCommitParents}"
-    echo "Local merge parents: ${headCommitParents}"
-    echo 'Github merge parents and local merge parents do not match; PR was updated since Jenkins job started. Try re-running the job.'
+    echo "Jenkins mergecommitsha: ${pullRequest.mergeCommitSha}"
+    echo "GH remote mergecommitsha: ${mergeCommitId}"
```

Comment:
I'm wondering if the two commit hashes are the wrong way around w.r.t. the text. Can you please check?

Reply:
Hi @vlkl-sap , 
They are fine. 
We get GH remote mergeCommitId from this code - [GitUtils.groovy](https://github.com/SAP/jenkins-library/blob/dd71cb1b0d7b34b11e3789d68f206dd410275d84/src/com/sap/piper/GitUtils.groovy#L12)
And Jenkins mergecommitsha from variable pullRequest.mergeCommitSha

Reply:
ok, you mean local commit rather than using Jenkins mergecommitsha. That makes sense.